l1-contracts-foundry/script/DeployHyperchain.s.sol

```diff
@@ -0,0 +1,261 @@
+            priorityTxMaxPubdata: uint32(config.contracts.diamondInitPriorityTxMaxPubdata),
+            minimalL2GasPrice: uint64(config.contracts.diamondInitMinimalL2GasPrice)
+        });
+
+        DiamondInitializeDataNewChain memory initializeData = DiamondInitializeDataNewChain({
+            verifier: IVerifier(config.addresses.verifier),
+            verifierParams: verifierParams,
+            l2BootloaderBytecodeHash: bytes32(Utils.getBatchBootloaderBytecodeHash()),
+            l2DefaultAccountBytecodeHash: bytes32(Utils.readSystemContractsBytecode("DefaultAccount")),
+            priorityTxMaxGasLimit: config.contracts.priorityTxMaxGasLimit,
+            feeParams: feeParams,
+            blobVersionedHashRetriever: config.addresses.blobVersionedHashRetriever
+        });
+
+        Diamond.DiamondCutData memory initData = Diamond.DiamondCutData({
+            facetCuts: facetCuts,
+            initAddress: config.addresses.diamondInit,
+            initCalldata: abi.encode(initializeData)
+        });
+
+        IBridgehub bridgehub = IBridgehub(config.addresses.bridgehub);
+
+        vm.broadcast();
+        vm.recordLogs();
+        bridgehub.createNewChain({
+            _chainId: config.eraChainId,
+            _stateTransitionManager: config.addresses.stateTransitionProxy,
+            _baseToken: config.addresses.baseToken,
+            _salt: config.contracts.bridgehubCreateNewChainSalt,
+            _admin: msg.sender,
+            _initData: abi.encode(initData)
+        });
+        console.log("Hyperchain registered");
+
+        // Get new diamond proxy address from emitted events
+        Vm.Log[] memory logs = vm.getRecordedLogs();
+        address diamondProxyAddress;
+        for (uint256 i = 0; i < logs.length; i++) {
+            if (logs[i].topics[0] == STATE_TRANSITION_NEW_CHAIN_HASH) {
+                diamondProxyAddress = address(uint160(uint256(logs[i].topics[2])));
+                break;
+            }
+        }
+        if (diamondProxyAddress == address(0)) {
+            revert("Diamond proxy address not found");
+        }
+        config.addresses.newDiamondProxy = diamondProxyAddress;
+    }
+
+    function addValidators() internal {
+        ValidatorTimelock validatorTimelock = ValidatorTimelock(config.addresses.validatorTimelock);
+
+        vm.startBroadcast();
+        validatorTimelock.addValidator(config.eraChainId, config.contracts.validatorSenderOperatorCommitEth);
+        validatorTimelock.addValidator(config.eraChainId, config.contracts.validatorSenderOperatorBlobsEth);
+        vm.stopBroadcast();
+
+        console.log("Validators added");
+    }
+
+    function configureZkSyncStateTransition() internal {
+        IZkSyncStateTransition zkSyncStateTransition = IZkSyncStateTransition(config.addresses.newDiamondProxy);
+
+        vm.startBroadcast();
+        zkSyncStateTransition.setTokenMultiplier(1, 1);
```

Comment:
Can you also get the 1, 1 variables from config please? The 1 1 are just place holders for now. The name should be the value they set in the contracts (BaseTokenGasMultiplierNominator, BaseTokenGasMultiplierDenominator )

Reply:
Fixed